sdk/data/azcosmos/cosmos_client.go

```diff
@@ -75,10 +75,29 @@ func NewClient(endpoint string, cred azcore.TokenCredential, o *ClientOptions) (
 	if err != nil {
 		return nil, err
 	}
-	scope, err := createScopeFromEndpoint(endpointUrl)
-	if err != nil {
-		return nil, err
+
+	var scope []string
+
+	if o != nil && o.ClientOptions.Cloud.Services != nil {
+		if svcCfg, ok := o.ClientOptions.Cloud.Services[ServiceName]; ok && svcCfg.Audience != "" {
+			audience := svcCfg.Audience
+			if !strings.HasSuffix(audience, "/.default") {
```

Comment:
I don't think we should support the scenario where the user put the scope in where the audience was expected. I'd rather not check this and let it produce an error with the server if the combined audience and scope end up wrong

Reply:
updated the logic